src/main/java/io/jenkins/plugins/folderauth/FolderAuthorizationStrategyManagementLink.java

```diff
@@ -107,21 +107,14 @@ public String getDisplayName() {
      * Adds a {@link GlobalRole} to {@link FolderBasedAuthorizationStrategy}.
      *
      * @param request the request to create the {@link GlobalRole}
-     * @throws IOException           when unable to add Global role
      * @throws IllegalStateException when {@link Jenkins#getAuthorizationStrategy()} is
      *                               not {@link FolderBasedAuthorizationStrategy}
      */
     @RequirePOST
     @Restricted(NoExternalUse.class)
-    public void doAddGlobalRole(@JsonBody GlobalRoleCreationRequest request) throws IOException {
-        Jenkins jenkins = Jenkins.get();
-        jenkins.checkPermission(Jenkins.ADMINISTER);
-        AuthorizationStrategy strategy = jenkins.getAuthorizationStrategy();
-        if (strategy instanceof FolderBasedAuthorizationStrategy) {
-            ((FolderBasedAuthorizationStrategy) strategy).addGlobalRole(request.getGlobalRole());
-        } else {
-            throw new IllegalStateException(Messages.FolderBasedAuthorizationStrategy_NotCurrentStrategy());
-        }
+    public void doAddGlobalRole(@JsonBody GlobalRoleCreationRequest request) {
```

Comment:
Unfortunately `getAPI()` does not allow redirecting API calls to another class. But maybe `${JENKINS_URL}/plugin/folder-auth/api/addGlobalRole()` is OK as a link (or `${MANAGEMENT_LINK}`/api). it would worth trying it so that some APi documentation and API links could be integrated in a single place.

https://wiki.jenkins.io/display/JENKINS/Exposing+data+to+the+remote+API

Reply:
Is this a suggestion to expose the API methods through `${rootURL}/folder-auth/api/` and have a Jelly page describing the REST API methods like it is done for the `Jenkins` object? The API class right now has a private constructor and all the functions are static and synchronize on the `Class` object. The `ManagementLink` right now provides all the REST methods by invoking these static functions; the `AuthorizationStrategy` does *not* provide any.

Please correct me if I'm wrong.

Reply:
Oh right, sorry for confusion. Then it is fine as is